src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -139,10 +141,26 @@ public void createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
         Long time = room.getDuration();
 
-        List<Problem> problems = problemService.getProblemsFromDifficulty(room.getDifficulty(), room.getNumProblems());
-        game.setProblems(problems);
-        setStartGameTimer(game, time);
+        // If specific problems specified, use those instead of difficulty setting
+        List<Problem> problems = game.getProblems();
+        problems.addAll(room.getProblems());
+
+        // Fill remaining problems with random ones by difficulty
+        int remaining = room.getNumProblems() - problems.size();
+        if (remaining > 0) {
```

Comment:
Imo it might be slightly redundant since the RoomService already verifies that `room.getProblems() <= room.getNumProblems()` - however, if you think that it'd be the safer move to include it, lmk and I can add a quick check

Reply:
Yeah I actually do think it'd be safer to include it, because if we end up changing `RoomService.java` in the future I doubt we'll remember to update this method accordingly as well. Thoughts? I'm still open to not changing anything, but that's my 2 cents.

Reply:
Sounds good - done!